src/fetch-mock.js

```diff
@@ -91,19 +102,22 @@ function compileRoute (route) {
 			debug('constructing starts with string matcher for route: ' + route.name);
 			expectedUrl = expectedUrl.substr(1);
 			route.matcher = function (url, options) {
-				return matchMethod(options) && url.indexOf(expectedUrl) === 0;
+				var req = toRequest(url, options);
+				return matchMethod(req) && req.url.indexOf(expectedUrl) === 0;
```

Comment:
https://github.com/vbauerster/fetch-mock/blob/request-matcher/src/fetch-mock.js#L76 and L77 - just change all the `options` to `req`